web/src/engine/websites/decorators/CoreView.ts

```diff
@@ -108,42 +114,77 @@ export function MangasMultiPageCSS(paths = mangaPaths, query = queryMangas, quer
+        Common.ThrowOnUnsupportedDecoratorContext(context);
+        return class extends ctor {
+            public async FetchChapters(this: MangaScraper, manga: Manga): Promise<Chapter[]> {
+                return FetchChaptersMultiPageAJAXV1.call(this, manga, query, extractor);
+            }
+        };
+    };
+}
+
+/**
+ * A class decorator for extracting chapters using Coreview API. Use this when website uses endpoint 'readable_products'.
+ * @param this - A reference to the {@link MangaScraper} instance which will be used as context for this method
+ * @param query - A CSS selector used to extract chapters from the api answer (html from json)
+ * @param extractor - A function to extract id and title from queried elements
  */
-export async function FetchChaptersSinglePageCSS(this: MangaScraper, manga: Manga, query = queryChapters, extractor = ChapterExtractor): Promise<Chapter[]> {
-    const request = new Request(new URL(manga.Identifier, this.URI));
-    let data = await FetchCSS<HTMLLinkElement>(request, queryChaptersAtomFeed);
-    const uri = new URL(data[0].href, this.URI);
-    uri.searchParams.set('free_only', '0'); // 0: include non-free, 1: exclude non-free
-    data = await FetchCSS(new Request(uri), query);
-    return data.map(element => {
-        const { id, title } = extractor.call(this, element, manga);
-        return new Chapter(this, manga, id, title);
+export async function FetchChaptersMultiPageAJAXV1(this: MangaScraper, manga: Manga, queryChapters: string = DefaultQueryChapters, extractor = ChapterExtractor): Promise<Chapter[]> {
+    const jsonData = (await FetchCSS(new Request(new URL(manga.Identifier, this.URI)), 'script#episode-json')).shift().dataset.value;
+    const { readableProduct: { series: { id } } } = JSON.parse(jsonData) as JSONSerie;
+
+    const { html } = await FetchJSON<APIChaptersHTML>(new Request(new URL(`/api/viewer/readable_products?aggregate_id=${id}&number_since=99999&number_until=0&read_more_num=150&type=episode`, this.URI)));
+    const doc = new DOMParser().parseFromString(html, 'text/html');
+    return [...doc.querySelectorAll(queryChapters)].map(chapter => {
+        const chapterData = extractor.call(this, chapter);
+        const title = chapterData.title.replace(manga.Title, '').trim();
+        return new Chapter(this, manga, chapterData.id, !title ? chapterData.title: title);
     });
 }
 
 /**
- * A class decorator that adds the ability to extract all chapters for a given manga from this website using the given CSS {@link query}.
- * The chapters are extracted from the composed url based on the `Identifier` of the manga and the `URI` of the website.
- * @param query - A CSS query to locate the elements from which the chapter identifier and title shall be extracted
- * @param extractor - An Extractor to get chapter infos
+ * An extension method for extracting chapters using Coreview API. Use this when endpoint 'pagination_readable_products' is available. Multipaged.
```

Comment:
The `multi-paged` information is redundant as the decorator factory name already implies this information.